extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -895,20 +919,30 @@ static public function onChangesListInsertArticleLink($list, &$articleLink, &$s,
 				$title = $wm->getMetaTitle();
 				$titleText = $wm->getArticleTitle()->getPrefixedText();
 				$pageText = $wm->getMainPageText();
-				$class = '';
-
-				$articleLink = ' <a href="'.$link.'" class="'.$class.'" >'.$title.'</a> ' . wfMessage( static::getMessagePrefix( $rc->getAttribute( 'rc_namespace' ) ) . '-new-message', array( $titleText, $pageText ) )->parse();
-
-				# Bolden pages watched by this user
-				# Check if the user is following the thread or the board
-				$user = $app->wg->User;
-				if ( $wm->isWatched( $user ) || $wm->isWallWatched( $user ) ) {
-					$articleLink = '<strong class="mw-watched">'.$articleLink.'</strong>';
+				$attribs = [
+					'class' => '',
+					'href' => $link,
+					'title' => $title
+				];
+
+				$articleLink = ' ' . Html::element( 'a', $attribs, $title ) . ' ' . $list->msg( static::getMessagePrefix( $rc->getAttribute( 'rc_namespace' ) ) . '-new-message', $titleText, $pageText )->parse();
+
+				# VOLDEV-3: Bolden pages watched by this user
+				$user = $list->getUser();
+				# bolding on Watchlist can be removed, on RC, it can not
+				$isWatchlist = $list->getTitle()->equals( SpecialPage::getTitleFor( 'Watchlist' ) );
+				if ( $app->wg->ShowUpdatedMarker && !$user->isAnon()
```

Comment:
Looks like this is the same conditional as the one on L1348. Assuming I'm not missing a subtle difference between the two, let's use the same broken out function for both. It will cut down significantly on duplicated code.